### PR TITLE
add gptneox embeddings, fix phi2 inputs, also fix the casting

### PR DESCRIPTION
**Pull request by winglian**
follow-up to #1074

### CODE REVIEW

File: src/axolotl/utils/models.py [hunk 1]
```diff
@@ -627,7 +628,7 @@ def load_model(
         for name, module in model.named_modules():
```
Comment (NanoCode012) [inline src/axolotl/utils/models.py:625]: Does the above if statement need to check for phi/gpt neox? How about miXtral?

Comment (winglian) [reply src/axolotl/utils/models.py:625]: it shouldn't since it's reusing `embedding_modules` defined in the same function from line 591.

Comment (winglian) [reply src/axolotl/utils/models.py:625]: mistral, mixtral, and llama all share the same embedding layers, which is the default return value for `get_linear_embedding_layers(...)`

Comment (NanoCode012) [reply src/axolotl/utils/models.py:625]: Ah sorry, I meant this.

```
    if needs_fa2_dtype or (
        cfg.flash_attention
        and (cfg.is_llama_derived_model or cfg.is_mistral_derived_model)
    ):
```
    
GH didn't show this part.

Comment (winglian) [reply src/axolotl/utils/models.py:625]: ah, good catch, I've updated that now. thanks!

**Pull request merged**